File: android/src/main/java/com/sangilov/plugins/animatedsplash/AnimatedSplashPlugin.java
package com.sangilov.plugins.animatedsplash;

import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin; 



@CapacitorPlugin(name = "AnimatedSplash")
public class AnimatedSplashPlugin extends Plugin {

    @PluginMethod
public void hide(PluginCall call) {
    if (SplashActivity.instance != null) {
        int duration = call.getInt("duration", 450); 
        SplashActivity.instance.closeSplash(duration);
    }
    call.resolve();
}
}

File: ios/Sources/AnimatedSplashPlugin/AnimatedSplashPlugin.swift
import Foundation
import Capacitor
import UIKit

@objc(AnimatedSplashPlugin)
public class AnimatedSplashPlugin: CAPPlugin {
    
    @objc func hide(_ call: CAPPluginCall) {
        DispatchQueue.main.async {
            let duration = call.getInt("duration") ?? 450
            
            if let splashVC = SplashViewController.instance {
                splashVC.closeSplash(duration: duration)
            }
            
            call.resolve()
        }
    }
}